account_payment_promissory_note_partner_address/views/account_payment_views.xml

```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+
+    <record id="view_account_payment_form" model="ir.ui.view">
+        <field name="name">account.payment.form</field>
+        <field name="model">account.payment</field>
+        <field name="inherit_id" ref="account.view_account_payment_form"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='company_id']" position="after">
+                <field name="payment_partner_id"/>
```

Comment:
Check when this field should be modified, depending on e,g. payment state.
This field actually replaces `partner_id`... and according to

https://github.com/odoo/odoo/blob/4cc086d330b73514fbc64a7fcf22d8a7a9f1b691/addons/account/models/account_payment.py#L67

`partner_id` can only be modified in `draft` state. This seems to be very restricted, but making it always editable is too permissive. Please ask LR